README.md

```diff
@@ -50,6 +50,33 @@ A simple library to generate and retrieve OAuth2 tokens for use with Google Clou
     }
     ```
 
+## Google Compute Metadata
+
+Every compute instance stores its metadata on a metadata server.
+Goth can query this metadata server to fetch authentication credentials
+for a service account within the instance.
+
+To query the metadata server for an access token, you must configure
+the following:
+
+  * `url` must be set to the base url for the metadata server.
```

Comment:
I wonder: if `{:instance, account}` is given, should we automatically default the url or that'd be trying to be too clever?

The advantage is there's less copy-pasting for the users as they'd most likely always use the same exact metadata url (I never used it so I wouldn't know, though.) We could do the same thing for the lower level `Goth.Token.fetch/1`, then it's really super simple to use:

```elixir
iex> Goth.Token.fetch(credentials: {:instance, "default"})
```

(This reminds me that currently that function needs to explicitly be passed the url & scope, but I think it's fine to use the same defaults as in `Goth.start_link`, I'll make this change later.)

Here's how we could document this then, I think it doesn't sound too bad:

```markdown
## Google Compute Metadata

Every compute instance stores its metadata on a metadata server.
Goth can query this metadata server to fetch authentication credentials
for a service account within the instance.

To query the metadata server for an access token, you must set `:credentials`
to tuple `{:instance, String.t()}`. If such credentials are set, the `:url`
will automatically default to `https://metadata.google.internal`.
```

WDYT?

Reply:
Yeah, I did not add the "magic" because I was not sure we wanted it and also I was not sure we would want to test it.  So I am open to suggestions there :)

Reply:
yeah, testing it would be a little bit tricky, we'd still assert on the `/computeMetadata...` bit but we wouldn't test the the hostname was set to `metadata.google.internal` because then we couldn't use Bypass which then I guess we could introduce a mock but I'd rather avoid that. I wouldn't be super concerned if we didn't test that particular bit, that we use the correct hostname, though.

Coming back to wether we want to introduce this magic in the first place, it's your call :D If not today, we'd always be able to add it later if someone wanted it because it would be backwards compatible, if someone explicitly sets the `:url` we would never touch it.